// api/mockParts.js
export default [
    {
        id: "1",
        categoryId: "1",
        name: "Фиксатор спининга",
        description: "Фиксатор спининга, с креплением на транцевую плиту",
        price: 1100,
        image: "/uploads/fiksator-spininga.jpg",
        size: "124.4мм x 59.2мм x 8.97мм",
        weight: "360 г",
        color: "черный"
    },
    {
        id: "2",
        categoryId: "1",
        name: "Ручка РИБ",
        description: "Ручка РИБ, с намоткой",
        price: 250,
        image: "/uploads/ruchka-rib.jpg",
        size: "124.4мм x 59.2мм x 8.97мм",
        weight: "140 г",
        color: "черный"
    },
    {
        id: "3",
        categoryId: "4",
        name: "Рым леерный",
        description: "Рым леерный удлиненный",
        price: 175,
        image: "/uploads/rym-leerniy.jpg",
        size: "124.4мм x 59.2мм x 8.97мм",
        weight: "110 г",
        color: "черный"
    },
    {
        id: "4",
        categoryId: "5",
        name: "Рым ручка",
        description: "Рым ручка буксировочная",
        price: 300,
        image: "/uploads/rym-ruchka.jpg",
        size: "124.4мм x 59.2мм x 8.97мм",
        weight: "180 г",
        color: "черный"
    },
    {
        id: "5",
        categoryId: "3",
        name: "Рым якорный",
        description: "Рым якорный под 12 мм трос",
        price: 850,
        image: "/uploads/rym-yakorniy.jpg",
        size: "124.4мм x 59.2мм x 8.97мм",
        weight: "260 г",
        color: "черный"
    },
    {
        id: "6",
        categoryId: "2",
        name: "Тент носовой",
        description: "Тент носовой с ветрозащитным окном",
        price: 2200,
        image: "/uploads/tent-nosovoy.jpg",
        size: "124.4мм x 59.2мм x 8.97мм",
        weight: "600 г",
        color: "синий"
    }
]
